MobPlugin/src/main/java/eu/cloudservice/cloudmob/MobSelector.java

```diff
@@ -0,0 +1,376 @@
+package eu.cloudservice.cloudmob;
+
+import de.dytanic.cloudnet.api.CloudAPI;
+import de.dytanic.cloudnet.api.handlers.adapter.NetworkHandlerAdapter;
+import de.dytanic.cloudnet.bridge.CloudServer;
+import de.dytanic.cloudnet.lib.NetworkUtils;
+import de.dytanic.cloudnet.lib.server.ServerState;
+import de.dytanic.cloudnet.lib.server.info.ServerInfo;
+import eu.cloudnetservice.cloudmob.MobConfig;
+import eu.cloudnetservice.cloudmob.MobItemLayout;
+import eu.cloudnetservice.cloudmob.MobPosition;
+import eu.cloudnetservice.cloudmob.ServerMob;
+import eu.cloudservice.cloudmob.event.BukkitMobInitEvent;
+import eu.cloudservice.cloudmob.event.BukkitMobUpdateEvent;
+import eu.cloudservice.cloudmob.listener.MobListener;
+import eu.cloudservice.cloudmob.listener.v18_112.ArmorStandListener;
+import eu.cloudservice.cloudmob.util.ItemStackBuilder;
+import eu.cloudservice.cloudmob.util.ReflectionUtil;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.entity.*;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public final class MobSelector {
+
+    private static MobSelector instance;
+    private final Map<String, ServerInfo> servers = new ConcurrentHashMap<>();
+    private Map<UUID, Mob> mobs;
+    private MobConfig mobConfig;
+
+    public MobSelector(MobConfig mobConfig) {
+        instance = this;
+        this.mobConfig = mobConfig;
+    }
+
+    public static MobSelector getInstance() {
+        return instance;
+    }
+
+    public static MobPosition toPosition(String group, Location location) {
+        return new MobPosition(group,
+            location.getWorld().getName(),
+            location.getX(),
+            location.getY(),
+            location.getZ(),
+            location.getYaw(),
+            location.getPitch());
+    }
+
+    public Mob spawnMob(final MobConfig mobConfig, final UUID uuid, final ServerMob serverMob) {
+        Location location = toLocation(serverMob.getPosition());
+
+        if (!location.getChunk().isLoaded()) {
+            location.getChunk().load();
```

Comment:
We should not spawn mobs, when the given chunk is unloaded.
Delay that process to when the chunk is actually loaded and cleanup upon unloading